common/src/node.rs

```diff
@@ -77,6 +96,39 @@ impl Graph {
         Some((chunk, coords))
     }
 
+    /// Populates a chunk with the given voxel data and ensures that margins are correctly removed
+    /// if necessary.
```

Comment:
It means to turn all the margin blocks of a chunk into air.

Reply:
I've rephrased "remove margin" to "clear margin" to make this explanation more natural, and I've added additional documentation. With the addition of a few code changes to make margin clearing more explicit (instead of implicit on the conversion of a solid chunk to dense), this code should hopefully be far more readable now.